compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -207,18 +208,26 @@ public void save() {
             writeBuildFile(buildFilePath);
             writeDependencies();
         } else {
-            // build file exists
-            BuildJson buildJson = readBuildJson(buildFilePath);
-            // need to update Dependencies toml
-            writeDependencies();
-            // check whether last updated time is expired
-            if (shouldUpdate) {
-                // update build json file
+            try {
+                // build file exists
+                BuildJson buildJson = readBuildJson(buildFilePath);
+                // need to update Dependencies toml
+                writeDependencies();
+                if (buildJson != null) {
+                    // check whether last updated time is expired
+                    if (shouldUpdate) {
+                        // update build json file
+                        writeBuildFile(buildFilePath);
+                    } else {
+                        // only update build time
+                        buildJson.setLastBuildTime(System.currentTimeMillis());
+                        ProjectUtils.writeBuildFile(buildFilePath, buildJson);
+                    }
+                } else {
+                    writeBuildFile(buildFilePath);
+                }
+            } catch (JsonSyntaxException | IOException e) {
                 writeBuildFile(buildFilePath);
```

Comment:
Looks like this can end up on a ProjectException from the ProjectUtils.writeBuildFile method. Shall we modify the util to throw the IOException, catch and ignore here?